utils/backupview/test/test.py

```diff
@@ -0,0 +1,141 @@
+
+# Tests for the backupview utility.
+# Use pytest ./test.py to run.
+
+import pytest
+
+import os.path
+import sys
+import tempfile
+import pathlib
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+backupview_dir = os.path.abspath(os.path.join(script_dir, ".."))
+if backupview_dir not in sys.path:
+    sys.path.append(backupview_dir)
+from backupview import open_backup, FileInfo, ExtractionInfo
+
+
+def calculate_num_files_and_total_size(dir):
+    count = 0
+    total_size = 0
+    for root, _, files in os.walk(dir, topdown=False):
+        for name in files:
+            if name:
+                count += 1
+                total_size += os.path.getsize(os.path.join(dir, root, name))
+    return (count, total_size)
+
+
+###########################################################################################
+# Actual tests
+
+
+def test_backup_1():
+    with open_backup(os.path.join(script_dir, "test_backup_1.zip")) as b:
+        assert b.get_subdirs("/") == ["shards"]
+        assert b.dir_exists("/shards")
+        assert not b.file_exists("/shards")
+        assert b.get_subdirs("/shards/") == ["1"]
+        assert b.get_subdirs("/shards/1/replicas/1") == ["data", "metadata"]
+        assert b.get_subdirs("/shards/1/replicas/1/metadata/") == ["mydb"]
+        assert b.get_files_in_dir("/shards/1/replicas/1/metadata/") == ["mydb.sql"]
+        assert b.file_exists("/shards/1/replicas/1/metadata/mydb.sql")
+
+        assert b.get_subdirs("/shards/1/replicas/1/data/mydb/tbl1") == [
+            "all_0_0_0",
+            "all_1_1_0",
+            "all_2_2_0",
+            "all_3_3_0",
+        ]
+
+        assert b.get_files_in_dir("/shards/1/replicas/1/data/mydb/tbl1/all_0_0_0") == [
+            "checksums.txt",
+            "columns.txt",
+            "count.txt",
+            "data.bin",
+            "data.cmrk3",
+            "default_compression_codec.txt",
+            "metadata_version.txt",
+            "serialization.json",
+        ]
+
+        assert b.get_databases() == ["mydb"]
+        assert b.get_tables(database="mydb") == ["tbl1", "tbl2"]
+        assert b.get_tables() == [("mydb", "tbl1"), ("mydb", "tbl2")]
+
+        assert b.get_create_query(database="mydb").startswith("CREATE DATABASE mydb")
+
+        assert b.get_create_query(table="mydb.tbl1").startswith(
+            "CREATE TABLE mydb.tbl1"
+        )
+
+        assert b.get_create_query(table=("mydb", "tbl1")).startswith(
+            "CREATE TABLE mydb.tbl1"
+        )
+
+        assert b.get_create_query(database="mydb", table="tbl2").startswith(
+            "CREATE TABLE mydb.tbl2"
+        )
+
+        assert b.get_partitions(table="mydb.tbl2") == ["all"]
+        assert b.get_parts(table="mydb.tbl2") == [
+            "all_0_0_0",
+            "all_1_1_0",
+            "all_2_2_0",
+        ]
+
+        assert b.get_parts(table="mydb.tbl2", partition="all") == [
+            "all_0_0_0",
+            "all_1_1_0",
+            "all_2_2_0",
```

Comment:
I also added a test, I don't know if it's necessary to include it in our CI.
Because this script doesn't start ClickHouse at any point and it doesn't check how ClickHouse work in any way. It's only a test for my utility, that's all.

Reply:
My two cents: it'd be nice if the backup format didn't break tools like this one or at least if this tool was guaranteed to work via an integration test. This way it could be extended to support the use case described in #52190.

Reply:
Done, I've added this test to CI